### PR TITLE
Agent ID Buff

### PR DESCRIPTION
**Pull request by Geevies**
* The agent ID card now makes you invisible to the AI when electronic warfare is enabled.
* The price of the agent ID card have been upped to 4, from 3.
* Agent IDs now have a charge, which lasts for a few minutes. When the charge runs out, electronic warfare is disabled.

Credits to Loaf (the man of many names) from Nebula for teaching me this technique.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/mob/living/carbon/give.dm [hunk 1]
```diff
@@ -38,6 +38,7 @@
 		return
 
 	if(usr.unEquip(I))
-		target.put_in_hands(I) // If this fails it will just end up on the floor, but that's fitting for things like dionaea.
-		usr.visible_message(span("notice", "\The [usr] handed \the [I] to \the [target]."), span("notice", "You give \the [I] to \the [target]."))
-
+		I.on_give()
```
Comment (skull132) [inline code/modules/mob/living/carbon/give.dm:41]: You might want to pass the `usr` as `giver` and `target` as `target` to this function. For later extensibility.

File: code/modules/mob/living/carbon/give.dm [hunk 2]
```diff
@@ -38,6 +38,7 @@
 		return
 
 	if(usr.unEquip(I))
-		target.put_in_hands(I) // If this fails it will just end up on the floor, but that's fitting for things like dionaea.
-		usr.visible_message(span("notice", "\The [usr] handed \the [I] to \the [target]."), span("notice", "You give \the [I] to \the [target]."))
-
+		I.on_give(usr, target)
+		if(I) // if on_give deletes the item, we don't want runtimes below
```
Comment (skull132) [inline code/modules/mob/living/carbon/give.dm:42]: this should be `if (!QDELETED(I))`

**Comment by Chada-1** [comment]
This is amazing and I was actually asked by arrow to turn my camera PR idea into a thing exactly like this, and so I'm v. glad this is being done. I might still try to do the camera thing at some point, but this is definitely good enough for now until we see how the no bolting + this AI is for the game.

**Comment by Kaedwuff** [comment]
Can you recharge the card?

File: code/game/objects/items/weapons/cards_ids_syndicate.dm [hunk 3]
```diff
@@ -79,14 +100,51 @@
 		return STATUS_CLOSE
 	return ..()
 
+
+/obj/item/card/id/syndicate/throw_at()
+	..()
+	electronic_warfare = FALSE
+	check_obfuscation()
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/cards_ids_syndicate.dm:107]: Doesn't throw_at call dropped()

Comment (Geevies) [reply code/game/objects/items/weapons/cards_ids_syndicate.dm:107]: doesn't look like it, no

**Comment by Geevies** [comment]
> 
> 
> Can you recharge the card?

it passively recharges, yeah

**Comment by TehFlaminTaco** [comment]
4 tcs destroys my Voice Changer + Chameleon Kit + Agent ID + Uplink implant build. Not objectively a bad thing, but a thing.

File: code/modules/mob/living/carbon/give.dm (hunk 2 above)
Comment (Geevies) [inline code/modules/mob/living/carbon/give.dm:42]:
```suggestion
		if(!QDELETED(I)) // if on_give deletes the item, we don't want runtimes below
```

**Pull request merged**